cpp/examples/protobuf/CMakeLists.txt

```diff
@@ -32,3 +32,9 @@ target_compile_options(pointcloud_proto_lib PUBLIC -Wno-error=unused-parameter)
 
 add_executable(example_protobuf_writer writer.cpp)
 target_link_libraries(example_protobuf_writer ${CONAN_LIBS} pointcloud_proto_lib)
+
+# NOTE: reader does not depend on pointcloud_proto_lib because it reads definitions
+# dynamically out of the MCAP file.
```

Comment:
This generic dynamic reading is a good example to have. However, I think many users will want to use statically generated cpp proto classes that they are already using in their other code. Can we extend this example or make another example that uses a generated cpp class to deserialize (matching on schema name) instead of the generic gp::Message?